pdns/cachecleaner.hh

```diff
@@ -147,7 +147,27 @@ template <typename T> uint64_t pruneLockedCollectionsVector(vector<T>& maps, uin
     }
     totErased += erased;
   }
-
+  
+  toTrim = toTrim - totErased;
+
+  if (toTrim > 0) {
+    //Just throwing away the old stuff wasn't sufficient, start from the top
+    
+    for(auto& mc : maps) {
+      WriteLock wl(&mc.d_mut);
+      auto& sidx = boost::multi_index::get<2>(mc.d_map);
+      
+      //Is it possible for toThrowAway to be larger than sidx.size()? Better safe than sorry
+      uint64_t toThrowAway = std::min(toTrim / maps.size(), sidx.size());
```

Comment:
Perhaps check that `toThrowAway` is greater than 0 and `continue` otherwise?

Reply:
We just checked if toTrim > 0 
and 
toThrowAway = toTrim / maps.size()

So for toThrowAway to be < 1, toTrim needed to be < maps.size(). I'm not sure how large maps.size() can be, but
statistical chances for this seems to be very slim and nothing will break of we don't check. Are we sure we want an extra check?

Reply:
Unless `sidx.size()` is 0, meaning this shard is empty.

Reply:
Okay, I added a check to make sure toThrowAway > 0